private-templates-service/client/src/utils/requireAuthentication/requireAuthentication.tsx

```diff
@@ -5,23 +5,27 @@ import { UserType } from '../../store/auth/types';
 import { RootState } from "../../store/rootReducer";
 import { connect } from "react-redux";
 
+import { AuthResponse } from 'msal';
+
 interface AuthProps {
   isAuthenticated: boolean,
+  graphAccessToken?: AuthResponse,
   user?: UserType,
   authButtonMethod: () => Promise<void>
 }
 
 const mapStateToProps = (state: RootState) => {
   return {
     isAuthenticated: state.auth.isAuthenticated,
+    graphAccessToken: state.auth.graphAccessToken,
     user: state.auth.user
   };
 };
 
 const requireAuthentication = <P extends object>(Component: React.ComponentType<P>) => {
   return class AuthenticatedComponent extends React.Component<P & AuthProps> {
 
-    isAuth = () => this.props.isAuthenticated;
+    isAuth = () => this.props.isAuthenticated && this.props.graphAccessToken !== undefined;
```

Comment:
So the problem was that the graph access token wasn't in the redux store when the action to get names + profile pictures were dispatched?

Reply:
yup